nuxeo-features/preview/nuxeo-preview-core/src/main/java/org/nuxeo/ecm/platform/preview/adapter/ImagePreviewer.java

```diff
@@ -28,14 +29,30 @@
 import org.nuxeo.ecm.core.api.Blob;
 import org.nuxeo.ecm.core.api.Blobs;
 import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.platform.picture.api.PictureView;
+import org.nuxeo.ecm.platform.picture.api.adapters.MultiviewPicture;
 import org.nuxeo.ecm.platform.preview.api.PreviewException;
 import org.nuxeo.ecm.platform.web.common.vh.VirtualHostHelper;
 
+import static org.nuxeo.ecm.platform.picture.api.ImagingDocumentConstants.PICTURE_VIEWS_PROPERTY;
+
 /**
  * @author Alexandre Russel
  */
 public class ImagePreviewer extends AbstractPreviewer implements MimeTypePreviewer {
 
+    protected Blob getContentBlob(Blob original, DocumentModel doc) {
+        if (doc.getProperty(PICTURE_VIEWS_PROPERTY) != null) {
+            MultiviewPicture multiviewPicture = doc.getAdapter(MultiviewPicture.class);
```

Comment:
you shouldn't need the previous check. Normally you use doc.getAdapter() and check if it's null